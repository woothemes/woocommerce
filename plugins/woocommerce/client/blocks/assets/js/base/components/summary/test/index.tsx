/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import Summary, { SummaryProps } from '../index';

const allowedSource =
	'<p class="some-class">' +
	'Lorem ipsum dolor sit amet ' +
	'<h1 style="color: red;">Heading</h1>' +
	'<ul><li>List item</li></ul>' +
	'<img src="https://example.com/image.jpg" alt="Image" />' +
	'</p>';

const disallowedSource =
	'<p class="some-class">' +
	'Lorem ipsum dolor sit amet ' +
	'<script src="http://evil.com" />' +
	'<h1 style="color: red;">Heading</h1>' +
	'<ul><li>List item</li></ul>' +
	'<img src="https://example.com/image.jpg" alt="Image" height="100" width="100" />' +
	'<script>alert("Hello");</script>' +
	'</p>';

const getProps = ( source: string ) =>
	( {
		source,
		maxLength: 1000,
		countType: 'words',
		className: 'test-class',
	} as SummaryProps );

describe( 'Summary component', () => {
	it( 'renders rich HTML with the allowed tags and attributes', () => {
		const props = getProps( allowedSource );
		const { container } = render( <Summary { ...props } /> );

		expect( container ).toMatchSnapshot( allowedSource );
	} );

	it( 'omits disallowed tags and attributes', () => {
		const props = getProps( disallowedSource );
		const { container } = render( <Summary { ...props } /> );

		expect( container ).toMatchSnapshot( allowedSource );
	} );
} );
