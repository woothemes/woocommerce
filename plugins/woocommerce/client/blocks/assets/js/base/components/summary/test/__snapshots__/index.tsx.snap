// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary component omits disallowed tags and attributes: <p class="some-class">Lorem ipsum dolor sit amet <h1 style="color: red;">Heading</h1><ul><li>List item</li></ul><img src="https://example.com/image.jpg" alt="Image" /></p> 1`] = `
<div>
  <div
    class="test-class"
  >
    <p
      class="some-class"
    >
      Lorem ipsum dolor sit amet 
    </p>
    

  </div>
</div>
`;

exports[`Summary component renders rich HTML with the allowed tags and attributes: <p class="some-class">Lorem ipsum dolor sit amet <h1 style="color: red;">Heading</h1><ul><li>List item</li></ul><img src="https://example.com/image.jpg" alt="Image" /></p> 1`] = `
<div>
  <div
    class="test-class"
  >
    <p
      class="some-class"
    >
      Lorem ipsum dolor sit amet 
    </p>
    

    <h1
      style="color: red;"
    >
      Heading
    </h1>
    

    <ul>
      

      <li>
        List item
      </li>
      

    </ul>
    

    <p>
      <img
        alt="Image"
        src="https://example.com/image.jpg"
      />
    </p>
    

  </div>
</div>
`;
